File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.cpp
class Solution {
public:
    bool uniformArray(vector<int>& nums1) {
        int minOdd = INT_MAX;
        int minEven = INT_MAX;

        for(int num : nums1){
            if(num & 1){
                minOdd = min(minOdd , num);
            }
            else{
                minEven = min(minEven , num);
            }
        }

        if(minOdd == INT_MAX ||minEven ==INT_MAX){
            return true;
        }

        return minEven > minOdd;
    }
};

File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.py
class Solution:
    def uniformArray(self, nums1: list[int]) -> bool:
        min_odd = float('inf')
        min_even = float('inf')

        for num in nums1:
            if num & 1:
                min_odd = min(min_odd , num)
            else:
                min_even = min(min_even , num)
            
        # we found all one sided number (all even or all odd)
        if min_odd == float('inf') or min_even == float("inf"):
            return True
        
        # if min even is greater then odd then we can make all the values odd 
        return min_even > min_odd
        